WooCommerce/WooCommerceTests/ViewRelated/MainTabBarControllerTests.swift

```diff
@@ -0,0 +1,111 @@
+import TestKit
+import XCTest
+@testable import WooCommerce
+
+final class MainTabBarControllerTests: XCTestCase {
+    private var stores: StoresManager!
+    private var tabBarController: MainTabBarController!
+
+    override func setUp() {
+        super.setUp()
+        let mockAuthenticationManager = MockAuthenticationManager()
+        ServiceLocator.setAuthenticationManager(mockAuthenticationManager)
+        stores = DefaultStoresManager.testingInstance
+        ServiceLocator.setStores(stores)
+        tabBarController = UIStoryboard(name: "Main", bundle: nil).instantiateInitialViewController() as? MainTabBarController
+    }
+
+    override func tearDown() {
+        SessionManager.testingInstance.reset()
+        tabBarController = nil
+        stores = nil
+        super.tearDown()
+    }
+
+    func test_tab_view_controllers_are_not_empty_after_updating_default_site() {
+        // Arrange
+        // Trigger `viewDidLoad`
+        XCTAssertNotNil(tabBarController.view)
+
+        // Action
+        let siteID: Int64 = 134
+        stores.updateDefaultStore(storeID: siteID)
+
+        // Assert
+        XCTAssertEqual(tabBarController.viewControllers?.count, 4)
+        assertThat((tabBarController.viewControllers?[WooTab.myStore.visibleIndex()] as? UINavigationController)?.topViewController,
+                   isAnInstanceOf: DashboardViewController.self)
+        assertThat((tabBarController.viewControllers?[WooTab.orders.visibleIndex()] as? UINavigationController)?.topViewController,
+                   isAnInstanceOf: OrdersRootViewController.self)
+        assertThat((tabBarController.viewControllers?[WooTab.products.visibleIndex()] as? UINavigationController)?.topViewController,
+                   isAnInstanceOf: ProductsViewController.self)
+        assertThat((tabBarController.viewControllers?[WooTab.reviews.visibleIndex()] as? UINavigationController)?.topViewController,
+                   isAnInstanceOf: ReviewsViewController.self)
+    }
+
+    func test_tab_view_controllers_are_empty_after_deauthentication() {
+        // Arrange
+        // Trigger `viewDidLoad`
+        XCTAssertNotNil(tabBarController.view)
+
+        // Action
+        let siteID: Int64 = 134
+        stores.updateDefaultStore(storeID: siteID)
+        stores.deauthenticate()
+
+        // Assert
+        XCTAssertEqual(tabBarController.viewControllers?.count, 0)
+    }
+
+    func test_tab_view_controllers_contain_a_different_reviews_tab_after_updating_to_a_different_site() throws {
```

Comment:
Maybe: 

```suggestion
    func test_viewControllers_are_replaced_after_updating_to_a_different_site() throws {
```

Or something along those lines. Just wanted to highlight that it's not just “reviews” tab anymore. 😅

Reply:
oops I forgot to update the test case name after replacing all tab view controllers! updated in https://github.com/woocommerce/woocommerce-ios/pull/2850/commits/d8f33ecce5c04540fc3f7cb16f6be691e1013920